Bookworm Build All Raise SQL syntax error

I'm in "Running the Script" section. Passed "Bookworm init", got errors when run "bookworm build all". I'm building bookworm on Ubuntu 16.04 VM. One error I noticed is sql syntax error: _mysql_exceptions.ProgrammingError: (1064, "You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '-bin' at line 1")

Has Anyone seen this one before?

Traceback:
bookworm -l WARNING -d cgi-bin prep database_wordcounts
27/2016 20:54:56:ERROR:Query failed:
DROP TABLE IF EXISTS words

Traceback (most recent call last):
  File "/usr/local/bin/bookworm", line 9, in <module>
    load_entry_point('bookwormDB==0.4.0', 'console_scripts', 'bookworm')()
  File "/usr/local/lib/python2.7/dist-packages/bookwormDB-0.4.0-py2.7.egg/bookwormDB/manager.py", line 567, in run_arguments
    getattr(my_bookworm,args.action)(args)
  File "/usr/local/lib/python2.7/dist-packages/bookwormDB-0.4.0-py2.7.egg/bookwormDB/manager.py", line 238, in prep
    getattr(self,args.goal)()
  File "/usr/local/lib/python2.7/dist-packages/bookwormDB-0.4.0-py2.7.egg/bookwormDB/manager.py", line 379, in database_wordcounts
    Bookworm.load_word_list()
  File "/usr/local/lib/python2.7/dist-packages/bookwormDB-0.4.0-py2.7.egg/bookwormDB/CreateDatabase.py", line 224, in load_word_list
    db.query("""DROP TABLE IF EXISTS words""")
  File "/usr/local/lib/python2.7/dist-packages/bookwormDB-0.4.0-py2.7.egg/bookwormDB/CreateDatabase.py", line 109, in query
    self.connect()
  File "/usr/local/lib/python2.7/dist-packages/bookwormDB-0.4.0-py2.7.egg/bookwormDB/CreateDatabase.py", line 82, in connect
    cursor.execute("CREATE DATABASE IF NOT EXISTS %s" % self.dbname)
  File "/usr/local/lib/python2.7/dist-packages/MySQLdb/cursors.py", line 205, in execute
    self.errorhandler(self, exc, value)
  File "/usr/local/lib/python2.7/dist-packages/MySQLdb/connections.py", line 36, in defaulterrorhandler
    raise errorclass, errorvalue
_mysql_exceptions.ProgrammingError: (1064, "You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '-bin' at line 1")
/usr/local/lib/python2.7/dist-packages/bookwormDB-0.4.0-py2.7.egg/bookwormDB/etc/bookworm_Makefile:154: recipe for target '.bookworm/targets/database_wordcounts' failed
make: **\* [.bookworm/targets/database_wordcounts] Error 1

--- a/bookwormDB/manager.py
+++ b/bookwormDB/manager.py
@@ -235,7 +235,7 @@ class BookwormManager(object):
 
         That's a little groaty, I know.
         """
-        getattr(self,args.goal)(args)
+        getattr(self,args.goal)(cmd_args=args)
         
     def build(self,args):
         """
@@ -267,15 +267,19 @@ class BookwormManager(object):
         logging.info("Parsing jsoncatalog.txt")
         bookwormDB.MetaParser.ParseJSONCatalog()
         
-    def preDatabaseMetadata(self, args):
+    def preDatabaseMetadata(self, cmd_args=None, **kwargs):
         import bookwormDB.CreateDatabase
         Bookworm = bookwormDB.CreateDatabase.BookwormSQLDatabase()
         logging.info("Writing metadata to new catalog file...")
-        Bookworm.variableSet.writeMetadata()
+        if cmd_args:
+            compress = cmd_args.gzip
+        else:
+            compress = False
+        Bookworm.variableSet.writeMetadata(compress=compress)
 
         # This creates helper files in the /metadata/ folder.
 
-    def text_id_database(self):
+    def text_id_database(self, **kwargs):
         """
         This function is defined in Create Database.
         It builds a file at .bookworm/texts/textids.dbm
@@ -283,15 +287,15 @@ class BookwormManager(object):
         import bookwormDB.CreateDatabase
         bookwormDB.CreateDatabase.text_id_dbm()
         
-    def metadata(self):
+    def metadata(self, **kwargs):
         self.diskMetadata()
         self.preDatabaseMetadata()
 
-    def catalog_metadata(self):
+    def catalog_metadata(self, **kwargs):
         from bookwormDB.MetaParser import parse_initial_catalog
         parse_initial_catalog()
 
-    def guessAtFieldDescriptions(self):
+    def guessAtFieldDescriptions(self, **kwargs):
         import bookwormDB.CreateDatabase
         import json
         Bookworm = bookwormDB.CreateDatabase.BookwormSQLDatabase(self.dbname,variableFile=None)
@@ -329,7 +333,7 @@ class BookwormManager(object):
         import bookwormDB.configuration
         bookwormDB.configuration.create(ask_about_defaults=askk)
             
-    def database_metadata(self):
+    def database_metadata(self, **kwargs):
         import bookwormDB.CreateDatabase
 
         Bookworm = bookwormDB.CreateDatabase.BookwormSQLDatabase(self.dbname)
@@ -368,7 +372,7 @@ class BookwormManager(object):
                                jsonDefinition=args.field_descriptions)
 
 
-    def database_wordcounts(self):
+    def database_wordcounts(self, **kwargs):
         """
         Builds the wordcount components of the database. This will die
         if you can't connect to the database server.
@@ -530,8 +534,9 @@ def run_arguments():
 
     catalog_prep_parser = extensions_subparsers.add_parser("preDatabaseMetadata",
                                                            help=getattr(BookwormManager, "preDatabaseMetadata").__doc__)
+    catalog_prep_parser.add_argument("--gzip", action="store_true")
     
-    for prep_arg in ['text_id_database', 'catalog_metadata', 'database_metadata', 'database_wordcounts']:
+    for prep_arg in ['text_id_database', 'catalog_metadata', 'database_metadata', 'database_wordcounts', 'guessAtFieldDescriptions']:
         extensions_subparsers.add_parser(prep_arg, help=getattr(BookwormManager, prep_arg).__doc__)
 
     """

--- a/bookwormDB/manager.py
+++ b/bookwormDB/manager.py
@@ -235,7 +235,7 @@ class BookwormManager(object):
 
         That's a little groaty, I know.
         """
-        getattr(self,args.goal)()
+        getattr(self,args.goal)(args)
         
     def build(self,args):
         """
@@ -267,7 +267,7 @@ class BookwormManager(object):
         logging.info("Parsing jsoncatalog.txt")
         bookwormDB.MetaParser.ParseJSONCatalog()
         
-    def preDatabaseMetadata(self):
+    def preDatabaseMetadata(self, args):
         import bookwormDB.CreateDatabase
         Bookworm = bookwormDB.CreateDatabase.BookwormSQLDatabase()
         logging.info("Writing metadata to new catalog file...")
@@ -526,7 +526,13 @@ def run_arguments():
     word_db_parser = tokenization_subparsers.add_parser("word_db",help="Turn a list of tokens into a sorted set of number IDs, even if there are more distinct types than can fit in memory, by writing to disk.")
     ########## Build components
     extensions_parser = subparsers.add_parser("prep", help="Build individual components: primarily used by the Makefile.")
-    extensions_parser.add_argument("goal",help="The name of the target.")
+    extensions_subparsers = extensions_parser.add_subparsers(title="goal",help="The name of the target.", dest="goal")
+
+    catalog_prep_parser = extensions_subparsers.add_parser("preDatabaseMetadata",
+                                                           help=getattr(BookwormManager, "preDatabaseMetadata").__doc__)
+    
+    for prep_arg in ['text_id_database', 'catalog_metadata', 'database_metadata', 'database_wordcounts']:
+        extensions_subparsers.add_parser(prep_arg, help=getattr(BookwormManager, prep_arg).__doc__)
 
     """
     Some special functions

--- a/bookwormDB/variableSet.py
+++ b/bookwormDB/variableSet.py
@@ -617,24 +617,37 @@ class variableSet:
 
         return bookids
 
-    def writeMetadata(self,limit=float("Inf")):
+    def writeMetadata(self,limit=float("Inf"),compress=False):
         #Write out all the metadata into files that MySQL is able to read in.
         """
         This is a general purpose, with a few special cases for the primary use case that this is the
         "catalog" table that hold the primary lookup information.
         """
+        import gzip
         linenum = 1
         variables = self.variables
         bookids = self.anchorLookupDictionary()
 
-        metadatafile = open(self.originFile)
+        try:
+            # Try as gzipped file. Overhead is trivial.
+            metadatafile = gzip.GzipFile(self.originFile)
+            metadatafile.peek(0)
+        except:
+            metadatafile = open(self.originFile)
 
 
         #Open files for writing to
+        path = os.path.dirname(self.catalogLocation)
         try:
-            catalog = open(self.catalogLocation,'w')
-        except IOError:
-            os.makedirs(os.path.dirname(self.catalogLocation))
+            os.makedirs(path)
+        except OSError:
+            if not os.path.isdir(path):
+                raise
+
+        if compress:
+            self.catalogLocation = self.catalogLocation + '.gz'
+            catalog = gzip.GzipFile(self.catalogLocation, 'w')
+        else:
             catalog = open(self.catalogLocation,'w')
 
         for variable in [variable for variable in variables if not variable.unique]:

